From 2d5b4aa78f91ddc8d3f609a228590c6cd0c8d761 Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Tue, 11 Aug 2026 09:46:00 +0000
Subject: [PATCH] SystemUI: Use transparent background for QS footer

[Port to A16] FooterActions.kt refactored (blur branch, textButtonColors); transparency applied at backgroundColor and TextButtonColors.background.

Change-Id: I59061006b35ed8c37820eab4da26c77946cfa7ce
---
 .../android/systemui/qs/footer/ui/compose/FooterActions.kt   | 5 ++---
 .../SystemUI/res/drawable/qs_footer_actions_background.xml   | 2 +-
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/packages/SystemUI/compose/features/src/com/android/systemui/qs/footer/ui/compose/FooterActions.kt b/packages/SystemUI/compose/features/src/com/android/systemui/qs/footer/ui/compose/FooterActions.kt
index be5d95bd73b8..ff983ba94ea6 100644
--- a/packages/SystemUI/compose/features/src/com/android/systemui/qs/footer/ui/compose/FooterActions.kt
+++ b/packages/SystemUI/compose/features/src/com/android/systemui/qs/footer/ui/compose/FooterActions.kt
@@ -183,8 +183,7 @@ fun FooterActions(viewModel: FooterActionsViewModel, modifier: Modifier = Modifi
         launch { viewModel.settings.collect { settings = it } }
     }
 
-    val backgroundColor =
-        if (!notificationShadeBlur()) colorAttr(R.attr.underSurface) else Color.Transparent
+    val backgroundColor = Color.Transparent
     val backgroundAlphaValue = if (!notificationShadeBlur()) backgroundAlpha::value else ({ 0f })
     val contentColor = MaterialTheme.colorScheme.onSurface
     val backgroundTopRadius = dimensionResource(R.dimen.qs_corner_radius)
@@ -715,7 +714,7 @@ private object FooterActionsDefaults {
     fun textButtonColors(): TextButtonColors =
         TextButtonColors(
             content = colorAttr(R.attr.onShadeInactiveVariant),
-            background = colorAttr(R.attr.underSurface),
+            background = Color.Transparent,
             border = BorderStroke(1.dp, colorAttr(R.attr.shadeInactive)),
         )
 
diff --git a/packages/SystemUI/res/drawable/qs_footer_actions_background.xml b/packages/SystemUI/res/drawable/qs_footer_actions_background.xml
index a7e8762a2593..6bfa8f75de03 100644
--- a/packages/SystemUI/res/drawable/qs_footer_actions_background.xml
+++ b/packages/SystemUI/res/drawable/qs_footer_actions_background.xml
@@ -15,7 +15,7 @@
 -->
 <inset xmlns:android="http://schemas.android.com/apk/res/android">
     <shape>
-        <solid android:color="?attr/underSurface"/>
+        <solid android:color="@android:color/transparent"/>
         <corners android:topLeftRadius="@dimen/qs_corner_radius"
                  android:topRightRadius="@dimen/qs_corner_radius"/>
     </shape>
-- 
2.43.0

